src/main/java/redis/clients/jedis/Connection.java

```
@@ -350,16 +354,41 @@ private void initializeFromClientConfig(JedisClientConfig config) {
         auth(credentialsProvider);
       }
 
-      int dbIndex = config.getDatabase();
-      if (dbIndex > 0) {
-        select(dbIndex);
-      }
-      String clientName = config.getClientName();
-      if (clientName != null) {
-        // TODO: need to figure out something without encoding
-        clientSetname(clientName);
-      }
+      try {
+        List<CommandArguments> fireAndForgetMsg = new ArrayList<>();
+
+        int dbIndex = config.getDatabase();
+        if (dbIndex > 0) {
+          fireAndForgetMsg.add(new CommandArguments(Command.SELECT).add(Protocol.toByteArray(dbIndex)));
```

IMO, we should not ignore error of SELECT command.